### PR TITLE
Feature/history local storage

### PR DESCRIPTION
**Pull request by OmerMoveo**

### CODE REVIEW

File: front-dispatcher/src/features/SearchBar/HistoryDiv/HistoryDiv.tsx [hunk 1]
```diff
@@ -10,30 +15,53 @@ import {
...
   };
   return (
-    <StyledHistoryDiv>
+    <StyledHistoryDiv mobile={mobile}>
       <StyledHistoryTextDiv isTitle={true}>
         <StyledHistoryTitleTexts>
           {historyStrings.RecentSearches}
         </StyledHistoryTitleTexts>
-        <StyledHistoryTitleTexts clickable={true}>
+        <StyledHistoryTitleTexts clickable={true} onClick={clearAllHistory}>
```
Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/HistoryDiv/HistoryDiv.tsx]: you can write just clickable if its true

File: front-dispatcher/src/features/SearchBar/InputWithIcon/InputWithIcon.tsx [hunk 2]
```diff
@@ -13,37 +13,47 @@ import { useAppDispatch, useAppSelector } from "../../../store";
...
   mobile?: boolean;
   focused?: boolean;
+  setFocused: (value: React.SetStateAction<boolean>) => void;
 }
-export default function InputWithIcon({
-  handleOnFocus,
-  mobile,
-  focused,
-}: props) {
+export default function InputWithIcon({ setFocused, mobile, focused }: props) {
```
Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/InputWithIcon/InputWithIcon.tsx]: maybe Arrow function ? =>

Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/InputWithIcon/InputWithIcon.tsx]: maybe Arrow function ? =>

File: front-dispatcher/src/features/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -18,25 +18,37 @@ interface searchbarProps {
...
 const SearchBar = ({ fullScreen, mobileBackFC }: searchbarProps) => {
   const [focused, setFocused] = useState<boolean>(false);
+  const searchDivRef = useRef<HTMLDivElement>(null);
   const filterState = useAppSelector((state) => state.filters);
   const { endpoint } = filterState;
+  useEffect(() => {
+    const handleClickOutside = (event: Event) => {
+      if (
+        searchDivRef.current &&
+        !searchDivRef.current.contains(event.target as Node)
```
Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/SearchBar.tsx:28]: Can you explain me the meaning of event.target as Node ?

Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/SearchBar.tsx:28]: Can you explain me the meaning of event.target as Node ?

File: front-dispatcher/src/strings/strings.ts [hunk 4]
```diff
@@ -212,4 +213,5 @@ export const apiUrlsStrings = {
 };
 export const usefulStrings = {
   whiteSpace: "\xa0",
+  Div: "DIV",
```
Comment (cohaviyarden) [inline front-dispatcher/src/strings/strings.ts:219]: also here for what you need this string ?

Comment (cohaviyarden) [inline front-dispatcher/src/strings/strings.ts:219]: also here for what you need this string ?

File: front-dispatcher/src/features/SearchBar/SearchBar.stories.tsx [hunk 5]
```diff
@@ -15,7 +15,6 @@ export default {
 } as ComponentMeta<typeof StyledSearchBarDiv>;
 const Template: ComponentStory<typeof StyledSearchBarDiv> = (args) => (
```
Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/SearchBar.stories.tsx:16]: You need to pass the props here : SearchBarStory.args = {};

Comment (cohaviyarden) [inline front-dispatcher/src/features/SearchBar/SearchBar.stories.tsx:16]: You need to pass the props here : SearchBarStory.args = {};

File: front-dispatcher/src/features/SearchBar/SearchBar.tsx (hunk 3 above)
Comment (OmerMoveo) [reply front-dispatcher/src/features/SearchBar/SearchBar.tsx:28]: so typescript understands e.target is an element

File: front-dispatcher/src/strings/strings.ts (hunk 4 above)
Comment (OmerMoveo) [reply front-dispatcher/src/strings/strings.ts:219]: to differ between the users click target location and not cause a click to add it to history and search input

**Pull request merged**